examples/c/sim_wt_model_nx6.c

```diff
@@ -418,22 +418,58 @@ int main()
 		************************************************/
 
 		printf("\nxn: \n");
-		for (ii=0; ii<nx; ii++)
-			printf("%8.5f ", x_sim[nsim0*nx+ii]);
-		printf("\n");
+		// for (ii=0; ii<nx; ii++)
+		// 	printf("%8.5f ", x_sim[nsim0*nx+ii]);
+		// printf("\n");
+		d_print_e_mat(1, nx, &x_sim[nsim0*nx], 1);
 
 		double *S_forw_out;
 		S_forw_out = NULL;
 		if(opts->sens_forw){
 			S_forw_out = out->S_forw;
 			printf("\nS_forw_out: \n");
-			for (ii=0;ii<nx;ii++){
-				for (jj=0;jj<NF;jj++)
-					printf("%8.5f ", S_forw_out[jj*nx+ii]);
-				printf("\n");
-			}
+			d_print_e_mat(nx, NF, S_forw_out, nx);
+			// for (ii=0;ii<nx;ii++){
+			// 	for (jj=0;jj<NF;jj++)
+			// 		printf("%8.5f ", S_forw_out[jj*nx+ii]);
+			// 	printf("\n");
+			// }
+		}
+
+
+		if(opts->sens_adj){
+			double *S_adj_out = out->S_adj;
+			printf("\nS_adj_out: \n");
+			d_print_e_mat(1, nx+nu, S_adj_out, 1);
 		}
 
+		// debug adjoints
+		struct blasfeo_dmat S_forw_result;
+		struct blasfeo_dvec adjoint_seed;
+		struct blasfeo_dvec forw_times_seed;
+
+		int Sf_mem_size = blasfeo_memsize_dmat(nx, nx+nu);
+		int adj_s_mem_size = blasfeo_memsize_dvec(nx);
+		int check_mem_size = blasfeo_memsize_dvec(nx+nu);
+
+		void *Sf_mem = malloc(Sf_mem_size);
```

Comment:
When you work with BLASFEO matrices, you need to provide aligned memory, e.g. using the function v_zeros_align in BLASFEO to allocate memory. Otherwise you get seg fault in certain architectures.